pkg/icingadb/db.go

```diff
@@ -131,7 +131,7 @@ func (db *DB) BuildInsertIgnoreStmt(into interface{}) (string, int) {
 	switch db.DriverName() {
 	case driver.MySQL:
 		// MySQL treats UPDATE id = id as a no-op.
-		clause = "ON DUPLICATE KEY UPDATE id = id"
+		clause = fmt.Sprintf(`ON DUPLICATE KEY UPDATE "%s" = "%s"`, columns[0], columns[0])
```

Comment:
1. No, not really.
2. I still update the PK.

Reply:
1. The comment is not technically wrong, it would still be more informative now if it stated that setting any column to itself won't generate actual writes (if that's indeed the case).
2. Probably depends on field order in the structs, but quite likely yes in most if not all cases.

Reply:
1. And I haven’t verified that case. But it doesn’t matter as MySQL unlikely checks for hardcoded "id". It's (still) the PK both the comment and the clause are about.
2. Yes, and in actually all cases it's the PK.